### PR TITLE
add DidFailProvisionalLoad method

### LINKED ISSUE #1105: did-fail-load does not fire on web contents
This prints out 'finished' but not 'failed'

```js
app.on('ready', function() {
  mainWindow = new BrowserWindow({width: 1024, height: 768, center: true});
  mainWindow.loadUrl('file:///this/file/does/not/exist.txt');
  mainWindow.webContents.on('did-fail-load', function() {
    console.log('failed');
  });
  mainWindow.webContents.on('did-finish-load', function() {
    console.log('finished');
  });
});
```

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #1105

### CODE REVIEW

**Comment by paulcbetts** [comment]
:+1: :cool:

**Comment by zcbenz** [comment]
Thanks!

File: spec/api-browser-window-spec.coffee [hunk 1]
```diff
@@ -58,6 +58,11 @@ describe 'browser-window module', ->
         done()
       w.loadUrl 'about:blank'
 
+    it 'should emit did-fail-load event', (done) ->
+      w.webContents.on 'did-finish-load', ->
```
Comment (zcbenz) [inline spec/api-browser-window-spec.coffee]: I think this test is not testing `did-fail-load`?

Comment (deepak1556) [inline spec/api-browser-window-spec.coffee]: super horrible typo :( , fixed thanks!

**Comment by maxkueng** [comment]
Thank you! :bouquet:

**Comment by gabssnake** [comment]
Since what version is this merged ? Not always firing on `electron-prebuilt 0.36.7` for URL `http://0.0.0.0/`.

**Pull request merged**